### PR TITLE
feat: full sync on plugin initialisation 

### PR DESCRIPTION
**Pull request by husayt**
Full synchronise on start with initialState

I like this library for it's simplicity, but I stumbled on a frustrating issue. Basically, there is an issue with current initial state. Currently it is pretty much useless as it ignores the values in cookies and localstorage. Instead I needed to run full sync on initialization.

This change achieves the objective. Now we run `syncUniversal` for every key in `initialState` on plugin registration.

That allows for very neat usage pattern:
For example, if I have an initialState like the following in nuxt config:
```js
  storage: {
    initialState: { testParam: false }
  },
```
then in my component (with decorators) I can simply declare
```js
  @State(s => s.storage.testParam)
  testParam
```
then I get computed property `testParam` with whatever value it had on my last session and on change I just fire `this.$storage.setUniversal("testParam", newVal)` to get value saved

### CODE REVIEW

**Comment by husayt** [comment]
@pi0 can you please comment on this. I have used the library with this patch for over a month without issues.

File: lib/plugin.js [hunk 1]
```diff
@@ -2,9 +2,14 @@ import Storage from './storage'
 import { serialize } from 'uri-js';
 
 export default function nuxtUniversalStorage(ctx, inject) {
+  const options=<%= JSON.stringify(options, null, 2) %>
   // Create new instance of Storage class
-  const storage = new Storage(ctx, <%= JSON.stringify(options, null, 2) %>)
+  const storage = new Storage(ctx, options)
 
+// Synchronise initial values
```
Comment (farzadso) [inline lib/plugin.js]: I think it's better to use consistent spacing @husayt

**Comment by farzadso** [comment]
@pi0 What are your opinions about the changes?

File: lib/plugin.js (hunk 1 above)
Comment (husayt) [reply lib/plugin.js]: yes sure

File: lib/storage.js [hunk 2]
```diff
@@ -81,7 +81,6 @@ export default class Storage {
     if (this._useVuex) {
       const storeModule = {
         namespaced: true,
-        state: () => this.options.initialState,
```
Comment (pi0) [inline lib/storage.js]: Doesn't it break reactivity?

File: lib/plugin.js [hunk 3]
```diff
@@ -2,9 +2,14 @@ import Storage from './storage'
...
+  for(let key in options.initialState) {
+	storage.syncUniversal(key, options.initialState[key])
```
Comment (pi0) [inline lib/plugin.js]: Spacing is broken :(

File: lib/plugin.js [hunk 4]
```diff
@@ -2,9 +2,14 @@ import Storage from './storage'
 import { serialize } from 'uri-js';
 
 export default function nuxtUniversalStorage(ctx, inject) {
+  const options=<%= JSON.stringify(options, null, 2) %>
   // Create new instance of Storage class
-  const storage = new Storage(ctx, <%= JSON.stringify(options, null, 2) %>)
+  const storage = new Storage(ctx, options)
```
Comment (pi0) [inline lib/plugin.js]: We are passing `options.initialState` here as well. Can't we refactor it to a class function like `syncInitialState`?

**Comment by pi0** [comment]
@husayt sorry for late review and thanks for PR :heart: This seems a really good enhancement.

**Comment by husayt** [comment]
Thanks @pi0 , indeed I have been using this for last few months without any issues

File: lib/plugin.js (hunk 4 above)
Comment (husayt) [reply lib/plugin.js]: Not sure what you mean. Lets get the initial version in and then we can improve this further. This has been waiting for so long I hardly remember the context.

File: lib/storage.js (hunk 2 above)
Comment (husayt) [reply lib/storage.js]: I don't think so. It works ok for me, but I might have missed something

File: lib/plugin.js (hunk 3 above)
Comment (husayt) [reply lib/plugin.js]: This is fixed now

**Comment by husayt** [comment]
@pi0 if all is ok, let's merge this and then we can discuss the latest

File: lib/plugin.js [hunk 5]
```diff
@@ -2,9 +2,14 @@ import Storage from './storage'
 import { serialize } from 'uri-js';
 
 export default function nuxtUniversalStorage(ctx, inject) {
+  const options=<%= JSON.stringify(options, null, 2) %>
```
Comment (farzadso) [inline lib/plugin.js]: Please use consistent spacing : 

`const options=<%= JSON.stringify(options, null, 2) %>`

👇

`const options = <%= JSON.stringify(options, null, 2) %>`

File: lib/storage.js (hunk 2 above)
Comment (farzadso) [reply lib/storage.js]: Why has this been removed in the first place? I agree with @pi0 , this does seem to break reactivity with no initial state.

File: lib/plugin.js (hunk 4 above)
Comment (husayt) [reply lib/plugin.js]: Hi @pi, please  check the latest, `initialState` is now being handled by `_initState`

File: lib/storage.js (hunk 2 above)
Comment (husayt) [reply lib/storage.js]: @pi0 and @farzadso I see what you mean. Latest should resolve this.

File: lib/plugin.js (hunk 5 above)
Comment (husayt) [reply lib/plugin.js]: Cool. This has also been resolved. Thanks for pointing out

**Comment by husayt** [comment]
> Can you please update the `README.md` file to include a line or two of documentation for this change?

That is also done and if no other issues then this can be merged

/cc @farzadso @pi0

File: lib/storage.js [hunk 6]
```diff
@@ -81,7 +81,7 @@ export default class Storage {
...
+        state: () => {},
```
Comment (pi0) [inline lib/storage.js]: This makes state `undefined`:

```js
const state = () => {}
console.log(state()) // undefined
```

Should be `() => ({})`

Comment (husayt) [reply lib/storage.js]: Ups, my bad.  Fixed it. Thanks for pointing out.

**Comment by farzadso** [comment]
@pi0 Pooya, do you have any other comments on this merge request?

**Comment by pi0** [comment]
@farzadso LGTM :)

**Comment by husayt** [comment]
@pi0 @farzadso  yes, definetely needs major version bump.

**Comment by farzadso** [comment]
@pi0 Just tested it on alibaba ... looks good. Will you do the merge please?

**Pull request merged**